### ISSUE TITLE
Simple query syntax errors when only a negative term is provided

### ISSUE THREAD

**Issue by dainiusjocas** [vespa-engine/vespa#29930]
**Describe the bug**
Simple query syntax errors when only a negative query term provided.

**To Reproduce**
1. Used sample application '[colbert](https://github.com/vespa-engine/sample-apps/tree/master/colbert)' but any application that has a field  of type string with an index has the same behavior.
2. Start Vespa, deploy app, feed documents.
3. Query `vespa query 'select * from sources * where userInput(@query)' query='-chunk:"water"'`
4. Error
```
{
    "root": {
        "id": "toplevel",
        "relevance": 1.0,
        "fields": {
            "totalCount": 0
        },
        "errors": [
            {
                "code": 4,
                "summary": "Invalid query parameter",
                "message": "Could not create query from YQL: Parsing '-chunk:\"water\"' only resulted in NullItem.",
                "stackTrace": "java.lang.IllegalArgumentException: Parsing '-chunk:\"water\"' only resulted in NullItem.\n\tat com.yahoo.search.yql.YqlParser.parseUserInput(YqlParser.java:876)\n\tat com.yahoo.search.yql.YqlParser.buildUserInput(YqlParser.java:819)\n\tat com.yahoo.search.yql.YqlParser.buildFunctionCall(YqlParser.java:386)\n\tat com.yahoo.search.yql.YqlParser.convertExpression(YqlParser.java:355)\n\tat com.yahoo.search.yql.YqlParser.buildTree(YqlParser.java:296)\n\tat com.yahoo.search.yql.YqlParser.parse(YqlParser.java:275)\n\tat com.yahoo.search.yql.MinimalQueryInserter.insertQuery(MinimalQueryInserter.java:95)\n\tat com.yahoo.search.yql.MinimalQueryInserter.search(MinimalQueryInserter.java:80)\n\tat com.yahoo.search.Searcher.process(Searcher.java:134)\n\tat com.yahoo.processing.execution.Execution.process(Execution.java:112)\n\tat com.yahoo.search.searchchain.Execution.search(Execution.java:499)\n\tat com.yahoo.prelude.searcher.FieldCollapsingSearcher.search(FieldCollapsingSearcher.java:96)\n\tat com.yahoo.search.Searcher.process(Searcher.java:134)\n\tat com.yahoo.processing.execution.Execution.process(Execution.java:112)\n\tat com.yahoo.search.searchchain.Execution.search(Execution.java:499)\n\tat com.yahoo.prelude.querytransform.PhrasingSearcher.search(PhrasingSearcher.java:60)\n\tat com.yahoo.search.Searcher.process(Searcher.java:134)\n\tat com.yahoo.processing.execution.Execution.process(Execution.java:112)\n\tat com.yahoo.search.searchchain.Execution.search(Execution.java:499)\n\tat com.yahoo.prelude.statistics.StatisticsSearcher.search(StatisticsSearcher.java:235)\n\tat com.yahoo.search.Searcher.process(Searcher.java:134)\n\tat com.yahoo.processing.execution.Execution.process(Execution.java:112)\n\tat com.yahoo.search.searchchain.Execution.search(Execution.java:499)\n\tat com.yahoo.search.handler.SearchHandler.searchAndFill(SearchHandler.java:348)\n\tat com.yahoo.search.handler.SearchHandler.search(SearchHandler.java:393)\n\tat com.yahoo.search.handler.SearchHandler.handleBody(SearchHandler.java:269)\n\tat com.yahoo.search.handler.SearchHandler.handle(SearchHandler.java:178)\n\tat com.yahoo.container.jdisc.ThreadedHttpRequestHandler.handle(ThreadedHttpRequestHandler.java:77)\n\tat com.yahoo.container.jdisc.ThreadedHttpRequestHandler.handleRequest(ThreadedHttpRequestHandler.java:87)\n\tat com.yahoo.container.jdisc.ThreadedRequestHandler$RequestTask.processRequest(ThreadedRequestHandler.java:191)\n\tat com.yahoo.container.jdisc.ThreadedRequestHandler$RequestTask.run(ThreadedRequestHandler.java:185)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:840)\n"
            }
        ]
    }
}
```
**Expected behavior**
Matches documents that doesn't contain a term `water` in the `chunk` field.

**Screenshots**
-

**Environment (please complete the following information):**
 - OS: [e.g. Docker]
 - Infrastructure: self-hosted
 - Versions [e.g. newest]

**Vespa version**
8.284.4

**Additional context**
Slack [thread](https://vespatalk.slack.com/archives/C038YTM5SUQ/p1705313359101479).

**Comment by bratseth** [comment]
Fixed by https://github.com/vespa-engine/vespa/pull/30004

**Comment by bratseth** [comment]
This will be released in 8.291
